parsec/mca/device/device.c

```diff
@@ -120,11 +128,12 @@ int parsec_get_best_device( parsec_task_t* this_task, double ratio )
     /* 0 is CPU, and 1 is recursive device */
     if( dev_index <= 1 ) {  /* This is the first time we see this data for a GPU, let's decide which GPU will work on it. */
         int best_index;
-        float weight, best_weight = parsec_device_load[0] + ratio * parsec_device_sweight[0];
+        parsec_device_module_t *dev = parsec_mca_device_get(0);
+        uint64_t duration, best_duration = time_estimate(this_task, dev);
```

Comment:
This is wrong, this will compute the shortest execution time for the task, not the best load balancing. Case in point device->device_load is only incremented/decremented, never read to consider if the device is overloaded. In most cases all GPU device will return the same time_estimate for the same task (since they have the same gflop/s capacity). 


Correct formula should consider `device->load+time_estimate(task,device))`

Reply:
problem has been resolved